src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -453,18 +455,18 @@ private string GetDebuggerDisplay()
         /// </summary>
         internal Diagnostic WithProgrammaticSuppression(ProgrammaticSuppressionInfo programmaticSuppressionInfo)
         {
-            Debug.Assert(this.ProgrammaticSuppressionInfo == null);
-            Debug.Assert(programmaticSuppressionInfo != null);
+            RoslynDebug.Assert(this.ProgrammaticSuppressionInfo == null);
+            RoslynDebug.Assert(programmaticSuppressionInfo != null);
 
             return new DiagnosticWithProgrammaticSuppression(this, programmaticSuppressionInfo);
         }
 
-        internal virtual ProgrammaticSuppressionInfo ProgrammaticSuppressionInfo { get { return null; } }
+        internal virtual ProgrammaticSuppressionInfo? ProgrammaticSuppressionInfo { get { return null; } }
 
         // compatibility
         internal virtual int Code { get { return 0; } }
 
-        internal virtual IReadOnlyList<object> Arguments
+        internal virtual IReadOnlyList<object>? Arguments
```

Comment:
>IReadOnlyList<object>? [](start = 25, length = 22)

It feels like `Arguments` should be non-`null`. Unfortunately, it looks `DiagnosticWithInfo.Arguments` may return `null`.

Reply:
Yep that's what forced my hand here. 

---
In reply to: [356891419](https://github.com/dotnet/roslyn/pull/40268#discussion_r356891419) [](ancestors = 356891419)